src/components/Card.js

```diff
@@ -1,22 +1,50 @@
+import Api from "./Api";
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "5e8acbbd-4426-43e4-895b-01bd99bee13b",
+    "Content-Type": "application/json",
+  },
+});
+
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick
+  ) {
     this.name = name;
     this.link = link;
+    this.id = _id;
+    this.isLiked = isLiked;
+    this.api = api;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
     this._cardElement = document
       .querySelector(this._cardSelector)
       .content.querySelector(".element")
       .cloneNode(true);
     this._cardImage = this._cardElement.querySelector(".element__picture");
+    this._likeBtn = this._cardElement.querySelector(".element__like-btn");
   }
 
   _handleLikeClick(evt) {
+    const element = evt.currentTarget.closest(".element");
     evt.currentTarget.classList.toggle("element__like-btn_active");
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.